// Copyright 2013 Google Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

// Author: morlovich@google.com (Maksim Orlovich)
// Protobufs to help snapshot shared memory caches to disk

syntax = "proto2";

option optimize_for = LITE_RUNTIME;

// If you need to make backwards incompatible changes to this proto, increase
// kSnapshotVersion in shared_mem_cache.cc.

package net_instaweb;

// NEXT ID: 4
message SharedMemCacheDumpEntry {
  required bytes raw_key = 1;
  required bytes value = 2;
  required sfixed64 last_use_timestamp_ms = 3;
};

// NEXT ID: 2
message SharedMemCacheDump {
  // Convention: more recently used entries are later in the array.
  repeated SharedMemCacheDumpEntry entry = 1;
};
